sql/core/src/test/resources/sql-tests/inputs/pgSQL/aggregates_part3.sql

```diff
@@ -0,0 +1,273 @@
+-- select string_agg(a,',') from (values(null),(null)) g(a);
+
+-- check some implicit casting cases, as per bug #5564
+-- select string_agg(distinct f1, ',' order by f1) from varchar_tbl;  -- ok
+-- select string_agg(distinct f1::text, ',' order by f1) from varchar_tbl;  -- not ok
+-- select string_agg(distinct f1, ',' order by f1::text) from varchar_tbl;  -- not ok
+-- select string_agg(distinct f1::text, ',' order by f1::text) from varchar_tbl;  -- ok
+
+-- Skip these tests because we do not have a bytea type
+-- string_agg bytea tests
+-- create table bytea_test_table(v bytea);
+
+-- select string_agg(v, '') from bytea_test_table;
+
+-- insert into bytea_test_table values(decode('ff','hex'));
+
+-- select string_agg(v, '') from bytea_test_table;
+
+-- insert into bytea_test_table values(decode('aa','hex'));
+
+-- select string_agg(v, '') from bytea_test_table;
+-- select string_agg(v, NULL) from bytea_test_table;
+-- select string_agg(v, decode('ee', 'hex')) from bytea_test_table;
+
+-- drop table bytea_test_table;
+
+-- [SPARK-27986] Support Aggregate Expressions with filter
+-- FILTER tests
+
+-- select min(unique1) filter (where unique1 > 100) from tenk1;
+
+-- select sum(1/ten) filter (where ten > 0) from tenk1;
+
+-- select ten, sum(distinct four) filter (where four::text ~ '123') from onek a
+-- group by ten;
+
+-- select ten, sum(distinct four) filter (where four > 10) from onek a
+-- group by ten
+-- having exists (select 1 from onek b where sum(distinct a.four) = b.four);
+
+-- [SPARK-28682] ANSI SQL: Collation Support
+-- select max(foo COLLATE "C") filter (where (bar collate "POSIX") > '0')
+-- from (values ('a', 'b')) AS v(foo,bar);
+
+-- outer reference in FILTER (PostgreSQL extension)
+select (select count(*)
+        from (values (1)) t0(inner_c))
+from (values (2),(3)) t1(outer_c);
+-- Rewriting to CASE WHEN will hit: Expressions referencing the outer query are not supported outside of WHERE/HAVING clauses
+-- select (select count(*) filter (where outer_c <> 0)
+--         from (values (1)) t0(inner_c))
+-- from (values (2),(3)) t1(outer_c);
+-- Rewriting to CASE WHEN will hit: Found an aggregate expression in a correlated predicate that has both outer and local references
+-- select (select count(inner_c) filter (where outer_c <> 0)
+--         from (values (1)) t0(inner_c))
+-- from (values (2),(3)) t1(outer_c); -- inner query is aggregation query
+-- Can not rewrite this pattern: aggregate_name(sub-query) filter()
+-- select
+--   (select max((select i.unique2 from tenk1 i where i.unique1 = o.unique1))
+--      filter (where o.unique1 < 10))
+-- from tenk1 o;					-- outer query is aggregation query
+-- subquery in FILTER clause (PostgreSQL extension)
+-- Rewriting to CASE WHEN will hit: IN/EXISTS predicate sub-queries can only be used in a Filter
+-- select sum(unique1) FILTER (WHERE
+--  unique1 IN (SELECT unique1 FROM onek where unique1 < 100)) FROM tenk1;
```

Comment:
From line 252 ~ 268, I understand the reason why you add comments like `Rewriting to CASE WHEN will hit: Expressions referencing the outer query are not supported outside of WHERE/HAVING clause`. But, let's have an explicit JIRA ID. Otherwise, let's not add the comment here.

Reply:
Yes. We should revert it to original query.